### ISSUE TITLE
Suppress Uglify’s 32000 characters warning

### ISSUE THREAD

**Issue by chasenlehara** [stealjs/steal-tools#716]
In a new DoneJS app, I get this warning when I build my app:

```
WARN: Output exceeds 32000 characters
```

This is harmless but disturbing to see when building a new app. I think it’d be nice if we suppressed this warning. I think it’s something we could work around by specifying Uglify’s `max-line-len` option.

**Comment by matthewp** [comment]
Why does the warning happen though? Are you sure it's not something that is bad and should be fixed elsewhere in the stack?

**Comment by chasenlehara** [comment]
I haven’t debugged the issue so I don’t know what there isn’t something bad happening elsewhere in the stack. From [reading](https://github.com/meteor/meteor/issues/8592) [similar](https://stackoverflow.com/questions/42568373/why-does-uglify-js-report-warn-output-exceeds-32000-characters) [issues](https://stackoverflow.com/questions/42577686/ember-production-build-error-warn-output-exceeds-32000-characters), I’m led to believe that the warning is generated when creating large bundle files with Uglify, since all of the code is on one line. I might be wrong though, so whoever looks into it will need to use their better judgment if it’s not a warning that should be suppressed. 😊

**Comment by matthewp** [comment]
Large bundles are potentially a bad thing, and can cause apps to load slower. If that's the reason, I don't want to suppress the warning. Need to look further into the issue though.

**Comment by m-mujica** [comment]
TL;DR

The only way to turn off the warning is by doing 

```js
var buildPromise = stealTools.build({}, {
  uglifyOptions: {
    output: {
      "max_line_len": false
    }
  }
});
```

Long version:

This warning is about the max length of a line when the code is minified. 

In 2.x which is the version steal-tools depends on is set to `3200` by default, there is a way to tell `uglifyjs` to beautify the output (will add a new line to split looong lines) but the line that triggers the warning is not easy to split:

```js
stealTools.build({}, {
  uglifyOptions: {
    output: {
      beautify: true // beautify the ugly code ¯\_(ツ)_/¯
    }
  }
});
```

![screen shot 2017-07-12 at 10 37 20](https://user-images.githubusercontent.com/724877/28129141-8a4ab072-66ef-11e7-8208-c1f8b83f0556.png)

That's the entire socket.io module in a single line :|  (I guess once we change transpile's globals we won't have this issue, at least not caused by steal-tools)

I'm not sure steal-tools should change uglify's default value for this property, unless we do it in a major version (and uglify 3 already defaults to false)

cc: @matthewp @chasenlehara

**Comment by matthewp** [comment]
Thanks @m-mujica. I believe you looked into what it would take to upgrade to uglify3 in the past, do you remember if that would be easy to do? I don't see why changing to uglify3 would be a major version change. We don't have any promise of what the output looks like.

**Comment by m-mujica** [comment]
@matthewp https://github.com/stealjs/steal-tools/pull/786

**Comment by chasenlehara** [comment]
@m-mujica, thank you so much for taking on this issue today! 🙏 Can this be made available as a patch release? I am happy to contribute towards any testing & releasing for this.

**Comment by chasenlehara** [comment]
Thank you for releasing this! https://github.com/stealjs/steal-tools/releases/tag/v1.6.0
